### DISCUSSION TITLE
Deep linking for the query

### DISCUSSION THREAD

**Discussion by PzKfWg** [microsoft-search/pnp-modern-search#2784] [Q&A] [answered]
Hi,

I understand how to configure the filters so that the selection gets in the URL:
https://microsoft-search.github.io/pnp-modern-search/usage/search-filters/#filter-deep-linking
But I can't seem to find how to do the same thing with the query part from the search box. 

I would like for the user that his entire query (search box and applied filters) be dynamically put in the current URL so that if he saves the URL as a favorite or reload the page, the whole query and filters are applied. Now, I can only figure out the filter part. How do I configure the search box web part so that any text put there gets also in the URL dynamically?

Thanks for this amazing web part!

**Comment by wobba** [comment, marked as answer]
This seems to work. Configure the search box like this. In the URL enter the URL to the same page you are on.

![image](https://user-images.githubusercontent.com/51104/220400593-f1702b76-abc6-4f4e-b8ae-bb2cccdceb08.png)

And then this setting.
![image](https://user-images.githubusercontent.com/51104/220400703-7db6eabb-4b10-4e04-85c4-1feb739bc338.png)

**Reply by PzKfWg** [reply]
Thank you so much, it works great (except that it adds a post-back for each new search but not a big deal!)

**Reply by wobba** [reply]
No postback happening on my test case - and entered the full URL with https://..../page.aspx.

**Reply by PzKfWg** [reply]
Indeed, I realize that it does not happen always. Only when a filter is selected and then I change the keywords : 
![post-back](https://user-images.githubusercontent.com/24944003/220468142-45874958-76e7-4e19-8567-133bb2147e3b.gif)
